app/src/hooks/reversiGame.ts

```diff
@@ -141,6 +157,26 @@ export const useReversiGame = (): ReversiGameType => {
     return whiteCount;
   };
 
+  // board と currentPlayer が変更されない限り同じ関数を返すメモ化された calculateMoveScores 関数
+  const memorizedCalculateMoceScores = useCallback(
+    () => calculateMoveScores(board, currentPlayer),
+    [board, currentPlayer],
+  );
+
+  // メモ化された関数が変更されたときに新たな盤面スコアがセットされる
+  useEffect(() => {
+    const scores = memorizedCalculateMoceScores();
+    setMoveScores(scores);
+  }, [memorizedCalculateMoceScores]);
+
+  // 盤面が変化した際に、評価値をセットする
+  useEffect(() => {
+    setboardEvaluatedScore({
+      white: evaluateBoard(board, "white"),
```

Comment:
imo
リテラル文字ではなく宣言している文字列を使用してほしいです。typoとか検出できます
```suggestion
      white: evaluateBoard(board, Disc.white),
```